packages/react-ui-core/src/Gmap/Markers.js

```diff
@@ -0,0 +1,69 @@
+import { Component } from 'react'
+import PropTypes from 'prop-types'
+import { setupMarker, removeMarker } from './utils/markerHelpers'
+
+const MARKER = ({ properties, geometry }) => ({
+  id: properties.id,
+  position: {
+    lat: geometry.coordinates[1],
+    lng: geometry.coordinates[0],
+  },
+  title: properties.id,
+  key: properties.id,
+})
+
+export default class Markers extends Component {
+  static propTypes = {
+    map: PropTypes.object,
+    marker: PropTypes.func,
+    geojson: PropTypes.object,
+  }
+
+  static defaultProps = {
+    marker: () => ({ }),
+  }
+
+  constructor(props) {
+    super(props)
+    this.markers = {}
+  }
+
+  componentWillUnmount() {
+    Object.keys(this.markers).map(id => {
+      removeMarker(this.markers[id])
+      delete this.markers[id]
+      return null
+    })
+  }
+
+  marker(feature) {
+    const props = this.props.marker(feature)
+    const defaults = MARKER(feature)
+
+    if (typeof props === 'object') {
+      return {
+        ...defaults,
+        ...props,
+      }
+    }
+
+    return defaults
+  }
+
+  renderMarker(map, feature) {
+    const id = feature.properties.id
+    const marker = this.marker(feature)
+
+    this.markers[id] = setupMarker(map, marker)
+    return this.markers[id]
+  }
+
+  render() {
+    const { map, geojson } = this.props
+
+    if (map) {
+      geojson.features.map(feature => this.renderMarker(map, feature))
```

Comment:
if I'm understanding this correctly, this just keeps appending to the visible markers when `geojson` updates? Rather than clear out existing markers?

Reply:
We should probably clear first... I'll add that in.